### PR TITLE
Fix #6343 handling posix not allowed on some Linux systems

### LINKED ISSUE #6343: installer fails, if posix is not installed on linux systems
installer fails on step 2 with this error:

Fatal error: Uncaught Error: Call to undefined function posix_getpwuid() in /SuiteCRM/include/utils.php:455 Stack trace: #0 /SuiteCRM/install/ready.php(284): getRunningUser() #1 /SuiteCRM/install.php(796): require('/html/SuiteCRM/...') #2 {main} thrown in /SuiteCRM/include/utils.php on line 455

This is because posix is not installed on an linux system with PHP-FPM up and running. The problem is, that there is only a check if the os is windows and if not, the script asumes, that posix is installed. We better check if posix is installed and not for windws in line 452:

 if (!function_exists('posix_getpwuid')) {

instead of

if (is_windows()) {

this should be fixed in update scripts as well.

### PR DESCRIPTION
**Pull request by pgorod**
## Description
According to the previous discussion on #6344, I examined 8 places in the code and decided to update 5 of them, the others aren't necessary.

## Instances of `posix` calls, and of places calling `getRunninguser`

1. https://github.com/salesagility/SuiteCRM/blob/master/modules/UpgradeWizard/systemCheck.php

2. https://github.com/salesagility/SuiteCRM/blob/master/modules/UpgradeWizard/uw_ajax.php

3. https://github.com/salesagility/SuiteCRM/blob/master/modules/UpgradeWizard/uw_utils.php

These 3 instances are only using this to output a message to the user. I guarded the code for the case where `posix` functions are disabled, and made sure the output made sense.

4. https://github.com/salesagility/SuiteCRM/blob/master/cron.php
Added an if before the others, so that a proper message is logged.

5. https://github.com/salesagility/SuiteCRM/blob/master/install/performSetup.php
No change done here. The call won't have any effect. When cron.php checks, it will report with the new log message added in 4.

6. https://github.com/salesagility/SuiteCRM/blob/master/install/ready.php
No change. Will output a generic message: `sudo crontab -e -u  <web_server_user>`. This was already there before the `allowed_cron_users` mechanism was introduced.

7. https://github.com/salesagility/SuiteCRM/blob/master/modules/Schedulers/Scheduler.php
Same as above.

8. https://github.com/salesagility/SuiteCRM/blob/master/include/utils.php
Added a check for the existence of the `posix` functions.

## How To Test This
For each situation below, test **with** and **without** `posix` functions active in Linux. This can be done with `disable_functions` in php.ini, or with `--disable-posix` flag in command-line.

1. Installation

2. Post-install, visit screens
  - Admin / Schedulers
  - Admin / Upgrade Wizard, run permissions check

3. Execute cron.php and check logs

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by pgorod** [comment]
@bjoern86 I'd appreciate it if you could help wit the testing of this PR. Thanks!

**Comment by Mac-Rae** [comment]
Assessed :+1:
